packages/server/src/course/course.controller.ts

```diff
@@ -121,10 +181,14 @@ export class CourseController {
       relations: ['staffList'],
     });
 
-    if (queues.some((q) => q.staffList.some((staff) => staff.id === user.id))) {
-      throw new UnauthorizedException(
-        ERROR_MESSAGES.courseController.checkIn.cannotCheckIntoMultipleQueues,
-      );
+    if (queues) {
+      if (
+        queues.some((q) => q.staffList.some((staff) => staff.id === user.id))
```

Comment:
Is there any reason why this is split into two ifs, instead of `if(queues && queues.some(...))` ?

Reply:
No, you're right I must have overlooked that simplification. I am updating it now! @IrisLiu-00